### DISCUSSION TITLE
[2023-11-06] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8231] [Weekly Demo Day]
@farmaazon

* Finished #8224 waits for final review and merge

https://github.com/enso-org/enso/assets/3919101/36dd4d5f-55d3-4f3c-ba49-e8425fae8270

@MichaelMauderer 

* Ready to merge: https://github.com/enso-org/enso/pull/8183
* Working on #8064

**Comment by vitvakatu** [comment]
Implementing file uploading:

- https://github.com/enso-org/enso/pull/8214
- https://github.com/enso-org/enso/pull/8227

https://github.com/enso-org/enso/assets/6566674/c0709d8f-3b95-4913-a6eb-caae225b6598

**Comment by wdanilo** [comment]
Website progress. The new website is ready to be tested! There are some bugs, but are small and fast to fix, you can find list of known issues on the [Discord channel here](https://discord.com/channels/401396655599124480/1171101124636446792/1171101124636446792).

https://github.com/enso-org/enso/assets/1623053/8c30051f-b676-4a04-a497-7ea55b78f906

**Comment by kazcw** [comment]
I'm designing a new AST and approach to editing and synchronizing the graph. A high-level overview is here: https://github.com/orgs/enso-org/discussions/8233; I will be finishing the design doc and starting implementation today.

**Comment by Frizi** [comment]
Continuing the work on widget implementation. Refactored the node information storage to use our reactive DB and indexes for easy access during widget selection. I have the fundamentals of widget loading and selection done. Next step is to add the ability to create widgets for elements that don't exist in AST yet (e.g. argument placeholders). https://github.com/enso-org/enso/issues/7732

https://github.com/enso-org/enso/assets/919491/e6888cd8-ab69-4ab5-863a-45d710566624

**Comment by mwu-tow** [comment]
* Fixed the macOS notarization https://github.com/enso-org/enso/pull/8192
* Alias analysis — going through final stages of the review https://github.com/enso-org/enso/issues/7863 / https://github.com/enso-org/enso/pull/8189
* Did some lookup into the python2 issue. https://github.com/enso-org/enso/pull/8209

**Comment by somebody1234** [comment]
- Reloading on changes and https://github.com/enso-org/enso/pull/8180
  - FS mock took a while, but will be very useful for my current task:
- (no PR yet) https://github.com/enso-org/enso/issues/8179

Video of viz reload working - note that the integrated code editor is emitting FS `Remove` and `Add` events:

https://github.com/enso-org/enso/assets/4046547/ce5da1ee-414f-4102-a5f5-3aa3f3407c55

**Comment by hubertp** [comment]
Relevant Engine update:
- Continued work on Inline Execution for Visualizations #8148
- Numerous refactorings in preparation to make loading of caches faster #8190, #8197, #8178, #8172
- Stability work #8225, #8201
